### PR TITLE
Timer

### PR DESCRIPTION
**Pull request by sebag90**
* added a 60 minutes inactivity timer to each bot. Once the time is over, the bot closes the room.
* enforced the int type to the attribute self.user of older bot for more consistency (api calls from the server return int user ids)

### CODE REVIEW

File: boxbot/boxbot.py [hunk 1]
```diff
@@ -106,6 +134,10 @@ def command(data):
             room_id = data["room"]
             game = self.game_per_room.get(room_id)
 
+            # snooze if the command does not coome from the bot
```
Comment (janagoetze) [inline boxbot/boxbot.py:137]: coome

File: boxbot/boxbot.py [hunk 2]
```diff
@@ -3,15 +3,38 @@
...
+        self.start_timer()
+
+    def start_timer(self):
+        self.timer = Timer(self.time*60, self.function, args=[self.room_id, self.game])
+        self.timer.start()
+
+    def snooze(self):
+        self.timer.cancel()
+        self.start_timer()
+        logging.debug("snooze")
```
Comment (janagoetze) [inline boxbot/boxbot.py:32]: this looks more like a restart. if it is, maybe rename? if it's not, please comment on how it works

File: clickbot/clickbot.py [hunk 3]
```diff
@@ -3,15 +3,38 @@
...
+        self.time = time
+        self.room_id = room_id
+        self.game = game
+        self.start_timer()
+
+    def start_timer(self):
+        self.timer = Timer(self.time*60, self.function, args=[self.room_id, self.game])
+        self.timer.start()
+
+    def snooze(self):
```
Comment (janagoetze) [inline clickbot/clickbot.py:29]: same here

File: echo/__main__.py [hunk 4]
```diff
@@ -1,16 +1,112 @@
...
+        self.function = function
+        self.time = time
+        self.room_id = room_id
+        self.start_timer()
+
+    def start_timer(self):
+        self.timer = Timer(self.time*60, self.function, args=[self.room_id])
+        self.timer.start()
+
+    def snooze(self):
```
Comment (janagoetze) [inline echo/__main__.py:23]: and here

**Review by janagoetze — COMMENTED** [review]
just the snooze looks odd to me

File: boxbot/boxbot.py (hunk 1 above)
Comment (sebag90) [reply boxbot/boxbot.py]: fixed

File: boxbot/boxbot.py (hunk 2 above)
Comment (sebag90) [reply boxbot/boxbot.py]: snooze was renamed to reset()

File: clickbot/clickbot.py (hunk 3 above)
Comment (sebag90) [reply clickbot/clickbot.py]: :+1:

File: echo/__main__.py (hunk 4 above)
Comment (sebag90) [reply echo/__main__.py]: :+1:

**Review by janagoetze — APPROVED** [review]

**Pull request merged**